inttest/customdomain/customdomain_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package customdomain
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/k0sproject/k0s/inttest/common"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CustomDomainSuite struct {
+	common.FootlooseSuite
+}
+
+func (s *CustomDomainSuite) TestK0sGetsUpWithCustomDomain() {
```

Comment:
Would be cool if we could check if the custom domain actually gets picked up.

Reply:
Any preferences how to do that properly?

Reply:
Not sure. Probably there are more clever ways than this, but I think Inspecting a pod's /etc/resolv.conf could work.

Reply:
@twz123 Added check for this in the last commit, PTAL